File: lib/models/city.dart
class Neighborhood {
  const Neighborhood({
    required this.id,
    required this.cityId,
    required this.name,
  });

  final int id;
  final int cityId;
  final String name;

  factory Neighborhood.fromJson(Map<String, dynamic> json) => Neighborhood(
    id: json['id'] as int,
    cityId: json['city_id'] as int,
    name: json['name'] as String,
  );
}

class City {
  const City({
    required this.id,
    required this.name,
    this.region,
    this.neighborhoods = const [],
  });

  final int id;
  final String name;
  final String? region;
  final List<Neighborhood> neighborhoods;

  factory City.fromJson(Map<String, dynamic> json) => City(
    id: json['id'] as int,
    name: json['name'] as String,
    region: json['region'] as String?,
    neighborhoods:
        (json['neighborhoods'] as List<dynamic>?)
            ?.map((e) => Neighborhood.fromJson(e as Map<String, dynamic>))
            .toList() ??
        const [],
  );
}

File: app/Models/City.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class City extends Model
{
    protected $fillable = ['name', 'region', 'latitude', 'longitude', 'is_active'];

    use HasFactory;

    protected function casts(): array
    {
        return [
            'latitude' => 'decimal:7',
            'longitude' => 'decimal:7',
            'is_active' => 'boolean',
        ];
    }

    public function neighborhoods(): HasMany
    {
        return $this->hasMany(Neighborhood::class);
    }

    public function products(): HasMany
    {
        return $this->hasMany(Product::class);
    }
}
